framework/src/transfers/MultiAppFieldTransferInterface.C

```diff
@@ -0,0 +1,396 @@
+      "The name of the Postprocessor in the from-app to evaluate an adjusting factor.");
+
+  std::vector<PostprocessorName> to_postprocessor = {"to_postprocessor"};
+  params.addParam<std::vector<PostprocessorName>>(
+      "to_postprocessor_to_be_preserved",
+      to_postprocessor,
+      "The name of the Postprocessor in the to-app to evaluate an adjusting factor.");
+  return params;
+}
+
+MultiAppFieldTransferInterface::MultiAppFieldTransferInterface(const InputParameters & parameters)
+  : MultiAppTransfer(parameters),
+    _from_var_name(getParam<std::vector<VariableName>>("source_variable")),
+    _to_var_name(getParam<std::vector<AuxVariableName>>("variable")),
+    _preserve_transfer(parameters.get<bool>("preserve_transfer")),
+    _from_postprocessor_to_be_preserved(
+        parameters.get<std::vector<PostprocessorName>>("from_postprocessor_to_be_preserved")),
+    _to_postprocessor_to_be_preserved(
+        parameters.get<std::vector<PostprocessorName>>("to_postprocessor_to_be_preserved")),
+    _use_nearestpoint_pps(false)
+{
+  if (_preserve_transfer)
+  {
+    /*
+     * Not sure how important to support multi variables
+     * Let us handle the single variable case only right now if the conservative capability is on
+     */
+    mooseAssert(_to_var_name.size() == 1 && _from_var_name.size() == 1,
+                " Support single variable only when the conservative capability is on ");
+    if (_direction == TO_MULTIAPP)
+    {
+      mooseAssert(_from_postprocessor_to_be_preserved.size() == _multi_app->numGlobalApps() ||
+                      _from_postprocessor_to_be_preserved.size() == 1,
+                  "Number of from Postprocessors should equal to the number of subapps, or use "
+                  "NearestPointIntegralVariablePostprocessor");
+      mooseAssert(_to_postprocessor_to_be_preserved.size() == 1,
+                  "Number of to Postprocessors should equal to 1");
+    }
+    else if (_direction == FROM_MULTIAPP)
+    {
+      mooseAssert(_from_postprocessor_to_be_preserved.size() == 1,
+                  "Number of from Postprocessors should equal to 1");
+      mooseAssert(_to_postprocessor_to_be_preserved.size() == _multi_app->numGlobalApps() ||
+                      _to_postprocessor_to_be_preserved.size() == 1,
+                  "Number of to Postprocessors should equal to the number of subapps, or use "
+                  "NearestPointIntegralVariablePostprocessor ");
+    }
+  }
+
+  mooseAssert(_to_var_name.size() == _from_var_name.size() && _to_var_name.size() >= 1,
+              " Number of target variable should equal to the number of source variables ");
+}
+
+void
+MultiAppFieldTransferInterface::initialSetup()
+{
+  if (_preserve_transfer)
+  {
+    if (_from_postprocessor_to_be_preserved.size() == 1 && _direction == TO_MULTIAPP)
+    {
+      FEProblemBase & from_problem = _multi_app->problemBase();
+      auto * pps = dynamic_cast<const NearestPointIntegralVariablePostprocessor *>(
```

Comment:
Remember we decided to have VectorPostprocessor but why here we are using `NearestPointIntegralVariablePostprocessor`?

Reply:
Really? I remembered last time we decided to use `NearestPointIntegralVariablePostprocessor`.  I could add one more support if it is necessary.

Reply:
Thought that pp is for Sebastian's model only. @friedmud mentioned few functions in the UserObject base class can be useful. I might be wrong. But I feel strange to let nearestpoint stuff control transfer preserving here. VectorPostprocessor is nice to user here but not a necessity.